src/private/InternalTextLink.tsx

```diff
@@ -73,7 +78,13 @@ export const InternalTextLink = ({ children, href }: Props) => {
   return (
     <TextLinkRenderer>
       {(rendererProps) => (
-        <HashLink {...rendererProps} scroll={scroll} smooth to={to}>
+        <HashLink
+          {...rendererProps}
+          onClick={(event) => event.preventDefault()}
```

Comment:
This is obviously too aggressive as it prevents page navigation; I suppose it should be limited to matching pathnames?